noir-projects/noir-contracts/contracts/app/nft_contract/src/types/nft_note.nr

```diff
@@ -202,12 +205,30 @@ pub struct PartialNFTNote {
 }
 
 impl PartialNFTNote {
-    pub fn commitment(self) -> Field {
-        self.commitment
+    /// Computes a partial note validity commitment. This commitment is stored in public storage when creating the
+    /// partial note in private and then is used to check that sender and note are valid upon partial note completion
+    /// in public.
+    pub fn compute_validity_commitment(self, sender: AztecAddress) -> Field {
+        let commitment_bytes: [u8; 32] = self.commitment.to_be_bytes();
+        let sender_bytes: [u8; 32] = sender.to_field().to_be_bytes();
+        // We use separator because the resulting commitment is used as a public storage slot in the partial note
+        // validity check scheme.
+        // (We use 4 bytes for the value because the constant is u32)
+        let separator_bytes: [u8; 4] =
+            (GENERATOR_INDEX__PARTIAL_NOTE_VALIDITY_COMMITMENT as Field).to_be_bytes();
+        // TODO(#10537): Can we use poseidon2 here given that partial_note.commitment() was computed with it? Or at
+        // least pedersen?
```

Comment:
@iAmMichaelConnor using sha256 because [the issue](https://github.com/AztecProtocol/aztec-packages/issues/10537) seems to apply here.

LMK if you think it's unnecessary here (or if you think at least pedersen can be used).

Reply:
Why would it apply? Because of repeated poseidon? I think the AES comment was more about shared secrets and assumptions by AES, and likely don't apply here. We already have nested poseidons in the trees, in the note hashes, etc.

Reply:
Yes, because of nested Poseidons. Will put there Poseidon once I'll manage to get confirmation from Mike.

Reply:
Mike is on vacation. Will just do it then.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/14379/commits/0a9edadf2789588a1340e72a061256b05769cb8b